packages/containers/src/DeleteResource/DeleteResource.container.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { componentState } from '@talend/react-cmf';
+import { ConfirmDialog } from '@talend/react-components';
+import { getActionsProps } from '../actionAPI';
+import deleteResourceConst from './deleteResource.constants';
+
+/**
+ * DeleteResource is used to delete a specific resource.
+ * When the component is mounted, it opens a confirm dialog.
+ * It uses the saga matching pattern to launch a race between the cancel and validate action.
+ */
+export default class DeleteResource extends React.Component {
+	static displayName = 'Container(DeleteResource)';
+	static propTypes = {
+		...componentState.propTypes,
+		'cancel-action': PropTypes.string.isRequired,
+		'validate-action': PropTypes.string.isRequired,
+		header: PropTypes.string,
+		uri: PropTypes.string.isRequired,
+		resourceType: PropTypes.string.isRequired,
+		// resourceInfo: PropTypes.shape({
+		// 	uri: PropTypes.string,
+		// 	resourceType: PropTypes.string,
+		// }).isRequired,
+	};
+	static contextTypes = {
+		registry: PropTypes.object.isRequired,
+		store: PropTypes.object.isRequired,
+	};
+
+	constructor(props, context) {
+		super(props, context);
+		this.getActions = this.getActions.bind(this);
+		this.getLabel = this.getLabel.bind(this);
+	}
+
+	/**
+	 * Get the label from the collections.
+	 * Return the label and a boolean to confirm that the item has been found.
+	 * @param {object} resourceInfo
+	 */
+	getLabel(resourceType, id) {
+		const resourcesCollection = this.context.store.getState().cmf.collections.get(resourceType);
+		if (resourcesCollection) {
+			const resource = resourcesCollection.find(
+				resoureCollection => resoureCollection.get('id') === id,
+			);
+			return resource ? { label: resource.get('label'), found: true } : {};
```

Comment:
Please use regular if/else.

Reply:
getLabel(resourceType, id) {
		const resourcesCollection = this.context.store.getState().cmf.collections.get(resourceType);
		if (resourcesCollection) {
			const resource = resourcesCollection.find(
				resoureCollection => resoureCollection.get('id') === id,
			);
		}
		return resource ? { label: resource.get('label'), found: true } : { label: '', found: false };
	}

Reply:
with a let resource at the beginning